src/R/Components/Impl/History/Implementation/RHistoryProvider.cs

```diff
@@ -16,18 +16,29 @@ namespace Microsoft.R.Components.History.Implementation {
     [Export(typeof(IRHistoryProvider))]
     internal class RHistoryProvider : IRHistoryProvider {
         private readonly ITextBufferFactoryService _textBufferFactory;
-        private readonly IFileSystem _fileSystem;
         private readonly IEditorOperationsFactoryService _editorOperationsFactory;
         private readonly IRtfBuilderService _rtfBuilderService;
         private readonly ITextSearchService2 _textSearchService;
         private readonly IRSettings _settings;
         private readonly IContentTypeRegistryService _contentTypeRegistryService;
         private readonly Dictionary<ITextBuffer, IRHistory> _histories;
 
+        private IFileSystem _fileSystem;
+        internal IFileSystem FileSystem {
```

Comment:
I guess it should be removed from the container then. Frankly I'd rather have containers use actual exported values rather than manually add stuff since it makes container != runtime container.